src/lightning/fabric/plugins/precision/precision.py

```diff
@@ -23,8 +23,10 @@
 from lightning.fabric.utilities.types import _PARAMETERS, Optimizable
 
 _PRECISION_INPUT_INT = Literal[64, 32, 16]
-_PRECISION_INPUT_STR = Literal["64", "32", "16", "bf16"]
-_PRECISION_INPUT = Union[_PRECISION_INPUT_INT, _PRECISION_INPUT_STR]
+_PRECISION_INPUT_STR_LEGACY_CONVERSION = {"64": "64-true", "32": "32-true", "16": "16-mixed", "bf16": "bf16-mixed"}
+_PRECISION_INPUT_STR_LEGACY = Literal["64", "32", "16", "bf16"]
```

Comment:
Nit: I wouldn't call it legacy, maybe alias? No big deal for the protected var here, more in general in the messaging.

Reply:
I originally named it legacy because they were discouraged. W.r.t the latest discussion, alias might in fact be better. Will change!